### DISCUSSION TITLE
是否可以为TextBox、ComboBox等编辑类控件单独增加label等附加属性

### DISCUSSION THREAD

**Discussion by nblfeng** [irihitech/Ursa.Avalonia#252] [Q&A] [answered]
目前可以在Form控件中对TextBox等控件设置Label等属性, 单无法独立使用, 在布局使用上灵活性有所不够~

![1](https://github.com/irihitech/Ursa.Avalonia/assets/20534153/ec544413-840e-452b-9c93-cff4f01d02ab)
![2](https://github.com/irihitech/Ursa.Avalonia/assets/20534153/fdc04abe-a415-43c7-8fd7-720a0c8dc92c)

**Comment by rabbitism** [comment]
可以描述一下使用场景吗？

**Comment by nblfeng** [comment]
> 可以描述一下使用场景吗？

比如系统的参数配置, 针对的是一些参数值的自定义, 在布局上类似上图附件, 参数项有占1列的, 也有占2列甚至更多列的情况.
在原生的控件中, 一般会采用TextBlock+TextBox来表示一个参数项的配置信息(配置项说明+配置项数值), 布局上会采用布局控件Grid、StackPanel等进行嵌套, 来达到排版对齐的效果.
Form控件能满足一部分需求, 但貌似无法实现布局中既占1列, 也能占用多列, 同时排版对齐的效果;
至于TextBox等编辑控件通过附加属性来增加文本的描述, 目前似乎也只能在Form控件中才能实现, 无法在脱离Form后达到类似的效果, 此处是参照了曾经WPF的第三方控件HandyControl中的TextBox给的建议.

**Comment by rabbitism** [comment, marked as answer]
首先，Form和FormGroup是支持自定义ItemsPanel的，可以参考我刚刚写的示例

https://github.com/irihitech/Avalonia-Tutorials/tree/main/Tutorials/FormLayout

![image](https://github.com/irihitech/Ursa.Avalonia/assets/14807942/bea6c8a1-c9cb-4178-96d7-33585752b4bc)


其次，在TextBox中存在InnerLeftContent，可以作为内部的label。

**Comment by nblfeng** [comment]
> 首先，Form和FormGroup是支持自定义ItemsPanel的，可以参考我刚刚写的示例
> 
> https://github.com/irihitech/Avalonia-Tutorials/tree/main/Tutorials/FormLayout
> 
> ![image](https://private-user-images.githubusercontent.com/14807942/331849526-bea6c8a1-c9cb-4178-96d7-33585752b4bc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTYxODM0MDksIm5iZiI6MTcxNjE4MzEwOSwicGF0aCI6Ii8xNDgwNzk0Mi8zMzE4NDk1MjYtYmVhNmM4YTEtYzljYi00MTc4LTk2ZDctMzM1ODU3NTJiNGJjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTIwVDA1MzE0OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJlZmMyNTMyYzVhNTM3ZGZlYWViNTA5MmJkY2ZjOGQ3OTMwMjkwZTE1ZWE2YmY3NTA0NDkxYjY0ODg0NWViNzYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.DMPEIZpP9jtiycCxUc_-QjJE2pTlrxawct47G7LkYYk)
> 
> 其次，在TextBox中存在InnerLeftContent，可以作为内部的label。

技术不精, 感谢分享~😁
